cpp/src/io/orc/writer_impl.cu

```diff
@@ -1019,6 +1027,9 @@ void writer::impl::write(table_view const &table)
       "be specified");
   }
 
+  auto device_columns = table_device_view::create(table);
+  rmm::device_uvector<column_device_view> device_leaf_columns(table.num_columns(), stream);
```

Comment:
This is here to store leaf columns so that a pointer to those can be stored in `stats_column_desc`. It's done this way because the `column_device_view` cannot be stored by value in `stats_column_desc` because its default ctor is deleted. And the pointer could not point to the child storage in parent's `column_device_view` because its `child` accessor returns by value.

Reply:
I can probably add this explanation as a comment.

Reply:
@vuule  Actually, device_leaf_columns **is** unused and I will remove it. This is a remnant of the PR which intended to overhaul the stats as well as ORC writer. When the ORC writer will be touched again so as to remove `nvstrdesc_s` then the device_uvector creation might be moved here.

`leaf_column_views` in `writer::impl::gather_statistics_blobs` is being used and created when `cudf::io::create_leaf_column_device_views` is called. In that function the `device_uvector` `leaf_column_views` is allocated and then the members of `stat_desc` are updated to point to elements in `leaf_column_views`.
Perhaps that is why it is confusing because the fact that `stats_desc` is being changed is not clear just from the function call. I can add this explanation to the documentation to make it clearer.